src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,147 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_APPLICATION_STATUS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COMPANY_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DURATION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROLE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_DATE;
+
+import java.util.Comparator;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.model.InternshipModel;
+import seedu.address.model.internship.Internship;
+import seedu.address.model.internship.InternshipComparators;
+
+/**
+ * Sorts the list of internships by a specified category.
+ */
+public class SortCommand extends InternshipCommand {
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts the list of internships by the specified "
+            + "category in either ascending (ASC) or descending (DESC) order. "
+            + "You should specify the category by using its prefix and then specify the order (either ASC or DESC).\n"
+            + "Available categories are:\n"
+            + PREFIX_COMPANY_NAME + " - Company Name\n"
+            + PREFIX_ROLE + " - Role\n"
+            + PREFIX_APPLICATION_STATUS + " - Application Status\n"
+            + PREFIX_DEADLINE + " - Deadline\n"
+            + PREFIX_START_DATE + " - Start Date\n"
+            + PREFIX_DURATION + " - Duration\n"
+            + "Format: " + COMMAND_WORD + " CATEGORY_PREFIX ORDER\n"
+            + "Example 1: " + COMMAND_WORD + " " + PREFIX_COMPANY_NAME + " ASC\n"
+            + "Example 2: " + COMMAND_WORD + " " + PREFIX_ROLE + " DESC";
+    public static final String MESSAGE_INVALID_COLUMN = "The value entered for the category is invalid";
+    public static final String MESSAGE_INVALID_ORDER = "The value entered for the order is invalid";
+    public static final String DEFAULT_KEYWORD = "default";
+
+    public final Prefix category;
+    /**
+     * Enumerates the possible sorting orders.
+     * <p>
+     * ASC - Ascending order (from smallest to largest).
+     * DESC - Descending order (from largest to smallest).
+     * </p>
+     */
+    public enum Order {
+        ASC,
+        DESC
+    }
+
+    public final Order sortedOrder;
+    public final String messageSuccess;
+    /**
+     * Creates a SortCommand to sort the internships by default of Company Name and Ascending Order.
+     */
+    public SortCommand() {
+        this(PREFIX_COMPANY_NAME, Order.ASC);
+    }
+
```

Comment:
I love the way you have this constructor which defaults an ascending sort by company name. Sounds like it will be very convenient for the user